components/chainhook-sdk/src/observer/mod.rs

```diff
@@ -1122,6 +1140,11 @@ pub async fn start_observer_commands_handler(
                     let mut total_occurrences: u64 = *chainhooks_occurrences_tracker
                         .get(&trigger.chainhook.uuid)
                         .unwrap_or(&0);
+                    // todo: this currently is only additive, and an occurrence means we match a chain event,
+                    // rather than the number of blocks. Should we instead add to the total occurrences for
+                    // every apply block, and subtract for every rollback? If we did this, we could set the
+                    // status to `Expired` when we go above `expire_after_occurrence` occurrences, rather than
+                    // deregistering
                     total_occurrences += 1;
```

Comment:
@lgalabru I can either make an issue for this if you think it makes sense, or remove the todo

Reply:
Created #413